apps/interop-rxjs-signal/src/app/list/photos.store.ts

```diff
@@ -30,106 +36,90 @@ const initialState: PhotoState = {
+export class PhotoStore {
+  private readonly photoService = inject(PhotoService);
 
-  private readonly endOfPage$ = this.select(
-    this.page$,
-    this.pages$,
-    (page, pages) => page === pages
-  );
+  // Signals
+  readonly state = signal(getSavedState());
+  readonly photos = computed(() => this.state().photos);
+  readonly page = computed(() => this.state().page);
+  readonly pages = computed(() => this.state().pages);
+  readonly error = computed(() => this.state().error);
+  readonly loading = computed(() => this.state().loading);
+  readonly endOfPage = computed(() => this.page() === this.pages());
+
+  // Sources
+  readonly searchControl = new FormControl({
+    value: getSavedState().search,
+    disabled: false,
+  });
 
-  readonly vm$ = this.select(
-    {
-      photos: this.photos$,
-      search: this.search$,
-      page: this.page$,
-      pages: this.pages$,
-      endOfPage: this.endOfPage$,
-      loading: this.loading$,
-      error: this.error$,
-    },
-    { debounce: true }
+  readonly search$ = this.searchControl.valueChanges.pipe(
+    startWith(this.searchControl.value),
+    filter((a) => !!a && a.length >= 3),
+    debounceTime(300),
+    distinctUntilChanged()
   );
 
-  ngrxOnStoreInit() {
-    const savedJSONState = localStorage.getItem(PHOTO_STATE_KEY);
-    if (savedJSONState === null) {
-      this.setState(initialState);
-    } else {
-      const savedState = JSON.parse(savedJSONState);
-      this.setState({
-        ...initialState,
-        search: savedState.search,
-        page: savedState.page,
-      });
-    }
-  }
+  readonly searchResults$ = this.search$.pipe(
+    map((search) => ({ search: search || '', page: 1 })), // TypeScript thinks it's string | null
+    switchMap(({ search, page }) =>
+      this.photoService.searchPublicPhotos(search, page).pipe(
+        tap(() => {
+          localStorage.setItem(
+            PHOTO_STATE_KEY,
+            JSON.stringify({ search, page })
+          );
+        }),
+        catchError((error: unknown) => of({ error }))
+      )
+    ),
+    share()
+  );
 
-  ngrxOnStateInit() {
-    this.searchPhotos(
-      this.select({
-        search: this.search$,
-        page: this.page$,
-      })
-    );
-  }
+  loadingChange$ = merge(
+    this.search$.pipe(map(() => true)),
+    this.searchResults$.pipe(map(() => false))
+  ).pipe(map((loading) => ({ ...this.state(), loading })));
 
-  readonly search = this.updater(
-    (state, search: string): PhotoState => ({
-      ...state,
-      search,
-      page: 1,
+  resultsChange$ = this.searchResults$.pipe(
+    map((result: { error: unknown } | FlickrAPIResponse) => {
+      const state = this.state();
+      if ('error' in result) {
+        return { ...state, error: JSON.stringify(result.error) };
+      } else {
+        return {
+          ...state,
+          photos: result.photos.photo,
+          pages: result.photos.pages,
+          error: '',
+        };
```

Comment:
Question: why don't you put this inside searchResults$ ?

Reply:
Oh. Maybe I could have. I'll look at this more.